n3fit/src/n3fit/model_trainer.py

```diff
@@ -472,6 +472,12 @@ def _model_generation(self, xinput, pdf_models, partition, partition_idx):
 
         if self.print_summary:
             training.summary()
+            pdf_model = training.get_layer("PDF_0")
+            pdf_model.summary()
+            nn_model = pdf_model.get_layer("NN_0")
+            nn_model.summary()
```

Comment:
If you'd loop over multiple seeds (in the other comment) the name here would be replica-dependent.